python/cudf_polars/cudf_polars/experimental/expressions.py

```diff
@@ -0,0 +1,484 @@
+
+        shuffle_on = (NamedExpr(next(names), child),)
+        input_ir = Shuffle(
+            input_ir.schema,
+            shuffle_on,
+            config_options,
+            input_ir,
+        )
+        partition_info[input_ir] = PartitionInfo(
+            count=pi.count,
+            partitioned_on=shuffle_on,
+        )
+
+    return input_ir, partition_info
+
+
+def _decompose_agg_node(
+    agg: Agg,
+    input_ir: IR,
+    partition_info: MutableMapping[IR, PartitionInfo],
+    config_options: ConfigOptions,
+    *,
+    names: Generator[str, None, None],
+) -> tuple[Expr, IR, MutableMapping[IR, PartitionInfo]]:
+    """
+    Decompose an Agg expression into multiple IR nodes.
+
+    Parameters
+    ----------
+    agg
+        The Agg node to decompose.
+    input_ir
+        The original input-IR node that ``agg`` will evaluate.
+    partition_info
+        A mapping from all unique IR nodes to the
+        associated partitioning information.
+    config_options
+        GPUEngine configuration options.
+    names
+        Generator of unique names for temporaries.
+
+    Returns
+    -------
+    expr
+        Decomposed Agg node.
+    input_ir
+        The rewritten ``input_ir`` to be evaluated by ``expr``.
+    partition_info
+        A mapping from unique nodes in the new graph to associated
+        partitioning information.
+
+    See Also
+    --------
+    _add_select_ir
+    _decompose_expr_node
+
+    Notes
+    -----
+    This function is called by ``_decompose_expr_node`` to decompose
+    an Agg node into multiple IR nodes. The new IR nodes are added
+    with ``_add_select_ir``.
+    """
+    expr: Expr
+    exprs: list[Expr]
+    if agg.name == "count":
+        # Chunkwise stage
+        columns, input_ir, partition_info = _add_select_ir(
+            [agg],
+            input_ir,
+            partition_info,
+            names=names,
+            repartition=True,  # Repartition
+        )
+
+        # Combined stage
+        (column,) = columns
+        columns, input_ir, partition_info = _add_select_ir(
+            [Agg(agg.dtype, "sum", None, column)],
+            input_ir,
+            partition_info,
+            names=names,
+        )
+        (expr,) = columns
+    elif agg.name == "mean":
+        # Chunkwise stage
+        exprs = [
+            Agg(agg.dtype, "sum", None, *agg.children),
+            Agg(agg.dtype, "count", None, *agg.children),
+        ]
+        columns, input_ir, partition_info = _add_select_ir(
+            exprs,
+            input_ir,
+            partition_info,
+            names=names,
+            repartition=True,
+        )
+
+        # Combined stage
+        exprs = [
+            BinOp(
+                agg.dtype,
+                plc.binaryop.BinaryOperator.DIV,
+                *(
+                    Agg(
+                        agg.dtype,
+                        "sum",
+                        None,
+                        column,
+                    )
```

Comment:
```suggestion
                    Agg(agg.dtype, "sum", None, column)
```

feels like it should fit the line length.